synapse/logging/context.py

```diff
@@ -695,16 +693,13 @@ def nested_logging_context(suffix: str) -> LoggingContext:
         )
         parent_context = None
         prefix = ""
```

Comment:
This is similar to the comment in `defer_to_threadpool`. I wonder if we should use `str(curr_context)` which will resolve to `"sentinel"` or the context's name.